### DISCUSSION TITLE
Not able to access json value by key

### DISCUSSION THREAD

**Discussion by RakshaC20** [nlohmann/json#4308] [Q&A]
I have a json string, jsonStr, as below:
[{“first”:123.321},{“second”:456.654},{“third”:789.987},{“fourth”:135.531},{“fifth”:246.642},{“sixth”:0}]
jsonStr is of json data type.

I am trying to access a particular value using the corresponding key but the code breaks with an exception e06d7363. 
The C++ code where the program breaks is:

std::cout << “jsonStr[first]: “ << jsonStr[“first”] << std::endl;

It looks pretty straightforward so I don’t understand where I am going wrong. Can someone please help me in this.
Let me know if more information is needed.

**Comment by gregmarr** [comment]
If that is your actual JSON text, then you have an array of single element objects, not a object with six elements.  As such, you need to access by array index first.

If your JSON text was this, then you could access it as you describe.

```
{“first”:123.321,“second”:456.654,“third”:789.987,“fourth”:135.531,“fifth”:246.642,“sixth”:0}

```

**Reply by RakshaC20** [reply]
Oh okay thank you! I am still a beginner exploring json. So now how do I access the values at arrays?
For example, if I want to access 456.654 based on its key, “second”, how do I do it? Could you please guide me?

**Reply by gregmarr** [reply]
You would need to know that it's in the second array position, if that's always true, or more likely, you'd need to iterate through all the items until you find the one that has the key you want.
https://www.godbolt.org/z/KxTW1989s

**Reply by RakshaC20** [reply]
The json data is a response from the http. And it can have variable number of key-value pairs so I will not be sure of the index. Is there any way to get the values without using indices?
Also, is there a way to get the values without using loops?

**Reply by gregmarr** [reply]
> The json data is a response from the http. And it can have variable number of key-value pairs so I will not be sure of the index.

I assumed that was the case.

> Is there any way to get the values without using indices? Also, is there a way to get the values without using loops?

There is no way to find a key in a sub object without iterating through the sub objects.  

People have asked about using JSON Path to do queries like this (`"$.[*].second"`).  It still ends up being iteration under the hood, it's just iteration that the library is doing instead of you.  It's also not supported by this library at this time.
